src/MooLite/plugins/WhisperManager/WhisperManagerPlugin.ts

```diff
@@ -0,0 +1,125 @@
+import { MooLitePlugin } from "src/MooLite/core/plugins/MooLitePlugin";
+import { MooLiteTab } from "src/MooLite/core/plugins/MooLiteTab";
+import { markRaw } from "vue";
+import WhisperManagerPluginDisplay from "src/MooLite/plugins/WhisperManager/WhisperManagerPluginDisplay.vue";
+import { ChatMessage } from "src/MooLite/core/chat/ChatMessage";
+import { ChatChannelTypeHrid } from "src/MooLite/core/chat/ChatChannelTypeHrid";
+import { ConversationMessage } from "./ConversationMessage";
+import { Game } from "src/MooLite/core/Game";
+import { Conversation } from "src/MooLite/plugins/WhisperManager/Conversation";
+import { PluginConfig } from "src/MooLite/core/plugins/config/PluginConfig";
+import { PluginConfigType } from "src/MooLite/core/plugins/config/PluginConfigType";
+
+export class WhisperManagerPlugin extends MooLitePlugin {
+    name: string = "Whisper Manager";
+    key = "whisper-manager";
+    description: string = "A plugin to help manage all of your conversations!";
+
+    _messages: ChatMessage[] = [];
+    _conversations: Record<string, Conversation> = {};
+    _activeConversationName: string = "";
+
+    config: PluginConfig[] = [
+        {
+            type: PluginConfigType.CheckBox,
+            name: "Hide System Messages",
+            description: "Hide system messages from the conversation list",
+            key: "hide-system-messages",
+            value: false,
+        },
+        {
+            type: PluginConfigType.CheckBox,
+            name: "Hide Mod Messages",
+            description: "Hide mod messages from the conversation list",
+            key: "hide-mod-messages",
+            value: false,
+        },
+    ];
+
+    tab: MooLiteTab = {
+        icon: "💬",
+        pluginName: this.name,
+        componentName: "WhisperManagerPluginDisplay",
+        component: markRaw(WhisperManagerPluginDisplay),
+        width: "w-80",
+    };
+
+    public get hideSystemMessages(): boolean {
+        return this.getConfig("hide-system-messages").value;
+    }
+
+    public get hideModMessages(): boolean {
+        return this.getConfig("hide-mod-messages").value;
+    }
+
+    public get activeConversation(): Conversation {
+        return this._conversations[this._activeConversationName];
+    }
+
+    public get activeConversationName(): string {
+        return this._activeConversationName;
+    }
+
+    public get messages(): ChatMessage[] {
+        return this._messages;
+    }
+
+    public get conversations(): Record<string, Conversation> {
+        let conversations: Record<string, Conversation> = this.hideSystemMessages
+            ? (({ System, ...o }) => o)(this._conversations)
+            : this._conversations;
+        conversations = this.hideModMessages ? (({ Mods, ...o }) => o)(conversations) : conversations;
```

Comment:
I have a hard time understanding what is going on here.
Is a [.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) not sufficient?

Reply:
No, because this is an object and not an array. This is the way to omit entries from an object without using lodash.